### ISSUE TITLE
Improve the method names of `Environment`

### ISSUE THREAD

**Issue by yutannihilation** [extendr/extendr#199]
Clippy complains at two points as the following (c.f. #196). For `new(<Environment>)` I wondered if we can

* rename `new(<Environment>)` to `with_parent(<Environment>)` and
* implement `new()` as `fn new() -> Self { Self::with_parent(empty_env()) }`

but then we'll need to rename `with_capacity()` to `with_parent_and_capacity()`, which feels a bit too verbose. I have no idea on `from_paris()` (something like `with_parent_and_pairs()`?). We need some investigation on how other crates have some nicer naming strategy here, or we can just ignore the warnings.

```
warning: methods called `new` usually take no self; consider choosing a less ambiguous name
  --> extendr-api/src/wrapper/environment.rs:17:16
   |
17 |     pub fn new(parent: Environment) -> Self {
   |                ^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention

warning: methods called `from_*` usually take no self; consider choosing a less ambiguous name
  --> extendr-api/src/wrapper/environment.rs:58:27
   |
58 |     pub fn from_pairs<NV>(parent: Environment, names_and_values: NV) -> Self
   |                           ^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention
```

**Comment by clauswilke** [comment]
Should we address this before a 0.3.0 release?

**Comment by andy-thomason** [comment]
Probably a good idea.

Thanks for spotting this, Hiroaki.

Andy.

**Comment by yutannihilation** [comment]
I just noticed the clippy warnings disappeared, and found this is already fixed by #205, thanks! :tada: 

We might come up with more regarding to naming, but let's discuss it again when it's needed. I'm closing this issue.
